source/slang/slang-shader-cache-index.cpp

```diff
@@ -0,0 +1,176 @@
+// slang-shader-cache-index.cpp
+#include "slang-shader-cache-index.h"
+
+#include "slang-hash-utils.h"
+#include "../core/slang-io.h"
+#include "../core/slang-string-util.h"
+#include "../core/slang-file-system.h"
+
+namespace Slang
+{
+
+ShaderCacheIndex::ShaderCacheIndex(SlangInt size, const char* path, ISlangFileSystem* fileSystem)
+{
+    entryCountLimit = size;
+    shaderCacheFileSystem = fileSystem;
+
```

Comment:
Did not initialize `indexFileName`. Not sure if you plan to use it.

Reply:
indexFilename is assigned in `loadCacheIndexFromFile`, not initializing it in the constructor is by design.